lib/main.js

```diff
@@ -28,77 +61,129 @@ export default {
       name: 'CSSLint',
       grammarScopes: ['source.css', 'source.html'],
       scope: 'file',
-      lintOnFly: true,
-      lint(textEditor) {
-        if (!helpers) {
-          helpers = require('atom-linter');
-        }
-        if (!path) {
-          path = require('path');
-        }
+      lintsOnChange: false,
+      lint: async (textEditor) => {
+        loadDeps();
         const filePath = textEditor.getPath();
         const text = textEditor.getText();
-        if (text.length === 0) {
-          return Promise.resolve([]);
+        if (!filePath || text.length === 0) {
+          // Empty or unsaved file
+          return [];
         }
-        const parameters = ['--format=json', '-'];
-        const exec = path.join(__dirname, '..', 'node_modules', 'atomlinter-csslint', 'cli.js');
+
+        const parameters = [
+          '--format=json',
```

Comment:
Note, requires CSSLint v1.0.0 minimum for this functionality. We might need a check for this if users start filing many support requests about broken versions.